piker/brokers/deribit/api.py

```diff
@@ -0,0 +1,717 @@
+        self,
+        pattern: str,
+        limit: int = 30,
+    ) -> dict[str, Any]:
+        data = await self.symbol_info()
+
+        matches = fuzzy.extractBests(
+            pattern,
+            data,
+            score_cutoff=35,
+            limit=limit
+        )
+        # repack in dict form
+        return {item[0]['instrument_name'].lower(): item[0]
+                for item in matches}
+
+    async def bars(
+        self,
+        symbol: str,
+        start_dt: Optional[datetime] = None,
+        end_dt: Optional[datetime] = None,
+        limit: int = 1000,
+        as_np: bool = True,
+    ) -> dict:
+        instrument = symbol
+
+        if end_dt is None:
+            end_dt = pendulum.now('UTC')
+
+        if start_dt is None:
+            start_dt = end_dt.start_of(
+                'minute').subtract(minutes=limit)
+
+        start_time = deribit_timestamp(start_dt)
+        end_time = deribit_timestamp(end_dt)
+
+        # https://docs.deribit.com/#public-get_tradingview_chart_data
+        resp = await self.json_rpc(
+            'public/get_tradingview_chart_data',
+            params={
+                'instrument_name': instrument.upper(),
+                'start_timestamp': start_time,
+                'end_timestamp': end_time,
+                'resolution': '1'
+            })
+
+        result = KLinesResult(**resp.result)
+        new_bars = []
+        for i in range(len(result.close)):
+
+            _open = result.open[i]
+            high = result.high[i]
+            low = result.low[i]
+            close = result.close[i]
+            volume = result.volume[i]
+
+            row = [
+                (start_time + (i * (60 * 1000))) / 1000.0,  # time
+                result.open[i],
+                result.high[i],
+                result.low[i],
+                result.close[i],
+                result.volume[i],
+                0
+            ]
+
+            new_bars.append((i,) + tuple(row))
+
+        array = np.array(new_bars, dtype=_ohlc_dtype) if as_np else klines
+        return array
+
+    async def last_trades(
+        self,
+        instrument: str,
+        count: int = 10
+    ):
+        resp = await self.json_rpc(
+            'public/get_last_trades_by_instrument',
+            params={
+                'instrument_name': instrument,
+                'count': count
+            })
+
+        return LastTradesResult(**resp.result)
+
+
+@acm
+async def get_client(
+    is_brokercheck: bool = False
+) -> Client:
+
+    if is_brokercheck:
+        yield Client
+        return
+
+    async with (
+        trio.open_nursery() as n,
+        open_autorecon_ws(_testnet_ws_url) as ws
+    ):
+
+        _rpc_id: Iterable = count(0)
+        _rpc_results: Dict[int, Dict] = {}
+
+        _expiry_time: int = float('inf')
+        _access_token: Optional[str] = None
+        _refresh_token: Optional[str] = None
+
+        def _next_json_body(method: str, params: Dict):
```

Comment:
this func is so thin you might want to just inline the `dict` in `json_rpc()` below.